bin/ch_core.c

```diff
@@ -370,6 +370,8 @@ enum img_type image_type(const char *ref, const char *storage_dir)
    struct stat st;
    FILE *fp;
    char magic[4];  // four bytes, not a string
+   char *conff = cat(ref, "/config.json");
+   char *rootfs = cat(ref, "/rootfs");
```

Comment:
Use `path_join()` for this (you may see `cat()` in the code but that’s obsolete).